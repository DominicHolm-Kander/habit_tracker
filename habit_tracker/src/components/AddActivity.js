import React from 'react'

export default function AddActivity() {
    return (
        <div>
            <form>
                    <label for="activityName">Name of activity
                    <input type="text" name="activityName"></input>
                    </label>
                    <label for="frequency">Frequency
                    <input type="text" name="frequency"></input>
                    </label>
                    <label for="priority">Priority (probs actually a radio button)
                    <input type="text" name="priority"></input>
                    </label>
                    <button type="button" id="addActivityButton" >Add an activity! (jsx needs updating)</button>
                </form>
        </div>
    )
}
