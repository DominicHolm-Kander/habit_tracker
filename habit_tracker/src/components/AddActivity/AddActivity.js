import React, { Component } from 'react'

export class AddActivity extends Component {
    state = {
        frequency: "Daily",
        isPriority: false
    }

    handleChangeOfActivityName = (event) => {
        this.setState({activityName: event.target.value})
    }
    handleChangeOfFrequency = (event) => {
        this.setState({frequency: event.target.value})
    }
    
    togglePriority = () => {
        if(this.state.isPriority){
            this.setState({isPriority : false})
        } else { 
            this.setState({isPriority: true})
        }
    }
    
    render() {
        return (
            <div>
            <form>
                <label for="activityName">Name of activity
                <input type="text" name="activityName" 
                onChange = {(event) => this.handleChangeOfActivityName(event)}>
                </input>
                </label>
                <label for="frequency">Frequency
                <select name="frequency" onChange = {(event) => this.handleChangeOfFrequency(event)}>
                    <option value = "Daily">Daily</option>
                    <option value = "Weekly">Weekly</option>
                    <option value = "Monthly">Monthly</option>
                </select>
                </label>
                <label for="priority">Priority (probs actually a radio button)
                <input type="checkbox" name="priority" onClick = {() => this.togglePriority()}/>This task is a priority
                </label>
                <button type="button" id="addActivityButton" onClick={() => this.props.addNewActivity(this.state)}>Add an activity! (jsx needs updating)</button>
            </form>
        </div>
        )
    }
}

export default AddActivity
