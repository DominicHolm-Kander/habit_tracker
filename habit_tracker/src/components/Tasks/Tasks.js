import React from 'react'
import Task from '../Task/Task'
import './Tasks.css'


export default function Tasks(props) {
    return (
        <div>
            <ul>
                {props.activitiesArray.map((activity, index) => {
                    return <Task className= {activity.completed ? "completedTask" : "incompleteTask"} activity={activity}/>
                })}
            </ul>
        </div>
    )
}
