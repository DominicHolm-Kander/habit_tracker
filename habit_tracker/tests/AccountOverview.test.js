import React from 'react'
import { shallow, mount, render } from 'enzyme'
import accountOverview from "../src/containers/AccountOverview"
import Tasks from '../src/components/Tasks/Tasks'
import waitUntil from "async-wait-until"; // For async testing of setState functions.
import regeneratorRuntime from 'regenerator-runtime';
import AddActivity from '../src/components/AddActivity/AddActivity';

let userInfo = {fn: 'text', ln: 'text'}
describe("shallow AccountOverview", () => {
    let wrapper;
    let displayAddActivityComponentMock = jest.fn()

    beforeEach(() => (wrapper = shallow(<accountOverview userInfo={userInfo}/>)))
    it("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })
    it("displayAddActivityComponent is called on click of add activity button", () => {
        wrapper.find('button').simulate('click');
        expect(displayAddActivityComponentMock).toHaveBeenCalled();
    })
    it('addActivity component is rendered when state of displayAddActivitycomponent is true',
    async () => {
        wrapper.setState({displayAddActivityComponent: true})
        await waitUntil(() => wrapper.instance().state.displayAddActivityComponent);
        expect(wrapper.containsMatchingElement(<AddActivity />)).toEqual(true)
    })
    it('addActivity button changes state correctly',
    async () => {
        wrapper.instance().displayAddActivityComponent()
        await waitUntil(() => wrapper.instance().state.displayAddActivityComponent)
        expect(wrapper.instance().state.displayAddActivityComponent).toEqual(true)
    })

})

describe('mounted AccountOverview', () => {
    let wrapper;
    let allActivities = [{
        username: 'jszrt',
        activityName: 'tennis',
        frequency: 'monthly',
        completed: false,
        isPriority: false,
        lastCompleted: '2019-12-16T14:34:43.432Z',
        streak: 0
    }]
    let displayAddActivityComponentMock = jest.fn()

    beforeEach(() => (wrapper = mount(<accountOverview userInfo={userInfo}/>)))
    it('Tasks component should only be rendered if activities are currently held in state',
    async () => {
        expect(wrapper.find(<Tasks />).length).toEqual(0)
        wrapper.setState({allActivities: allActivities})
        await waitUntil(() => wrapper.instance().state.allActivities);
        expect(wrapper.find(<Tasks />).length).toEqual(1);
    })
})





