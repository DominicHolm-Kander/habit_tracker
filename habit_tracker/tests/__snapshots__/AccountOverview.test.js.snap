// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow AccountOverview should render correctly 1`] = `
<accountOverview
  userInfo={
    Object {
      "fn": "text",
      "ln": "text",
    }
  }
/>
`;
